Kernel/Arch/aarch64/Time/ARMv8Timer.cpp

```diff
@@ -107,7 +107,7 @@ ErrorOr<void> ARMv8TimerDriver::probe(DeviceTree::Device const& device, StringVi
 {
     auto const interrupts = TRY(device.node().interrupts(DeviceTree::get()));
 
-    if (device.node().has_property("interrupt-names"sv) || interrupts.size() != 4)
+    if (device.node().has_property("interrupt-names"sv))
```

Comment:
Should we check for `>= 4`? Do we make any assumptions about this count, or is any count fine? If the latter, why did we check in the first place?

Reply:
Oh true, we should check `>= to_underlying(DeviceTreeTimerInterruptIndex::EL1Virtual)`, since that is the timer we want to use.

I originally thought that timer nodes without "interrupt-names" need to define 4 interrupts, but that is only true for the old version of the binding. See my commit message in 4cc9bea86dae6753f7d1e2c4b01c3977ee3f64f1.
But the new binding that was added in https://github.com/torvalds/linux/commit/a6cf39fbe133857a7f26f0cab8198869caf44dee allows interrupt properties with more or less entries even for nodes without "interrupt-names". I can't remember why I thought we should keep the old check as well.